File: decompiled/sources/com/hjq/toast/ActivityToast.java
package com.hjq.toast;

import android.app.Activity;
import android.view.View;
import android.widget.TextView;
import com.hjq.toast.config.IToast;

public final class ActivityToast implements IToast {
    private int mDuration;
    private int mGravity;
    private float mHorizontalMargin;
    private TextView mMessageView;
    private final ToastImpl mToastImpl;
    private float mVerticalMargin;
    private View mView;
    private int mXOffset;
    private int mYOffset;

    public TextView findMessageView(View view) {
        if (view instanceof TextView) {
            int id = view.getId();
            if (id == -1) {
                view.setId(16908299);
            } else if (id != 16908299) {
                throw new IllegalArgumentException("You must set the ID value of TextView to android.R.id.message");
            }
            return (TextView) view;
        }
        View findViewById = view.findViewById(16908299);
        if (findViewById instanceof TextView) {
            return (TextView) findViewById;
        }
        throw new IllegalArgumentException("You must include a TextView with an ID value of android.R.id.message");
    }

    public ActivityToast(Activity activity) {
        this.mToastImpl = new ToastImpl(activity, this);
    }

    public void show() {
        this.mToastImpl.show();
    }

    public void cancel() {
        this.mToastImpl.cancel();
    }

    public void setText(int i) {
        View view = this.mView;
        if (view != null) {
            setText((CharSequence) view.getResources().getString(i));
        }
    }

    public void setText(CharSequence charSequence) {
        TextView textView = this.mMessageView;
        if (textView != null) {
            textView.setText(charSequence);
        }
    }

    public void setView(View view) {
        this.mView = view;
        if (view == null) {
            this.mMessageView = null;
        } else {
            this.mMessageView = findMessageView(view);
        }
    }

    public View getView() {
        return this.mView;
    }

    public void setDuration(int i) {
        this.mDuration = i;
    }

    public int getDuration() {
        return this.mDuration;
    }

    public void setGravity(int i, int i2, int i3) {
        this.mGravity = i;
        this.mXOffset = i2;
        this.mYOffset = i3;
    }

    public int getGravity() {
        return this.mGravity;
    }

    public int getXOffset() {
        return this.mXOffset;
    }

    public int getYOffset() {
        return this.mYOffset;
    }

    public void setMargin(float f, float f2) {
        this.mHorizontalMargin = f;
        this.mVerticalMargin = f2;
    }

    public float getHorizontalMargin() {
        return this.mHorizontalMargin;
    }

    public float getVerticalMargin() {
        return this.mVerticalMargin;
    }
}

File: decompiled/sources/com/hjq/toast/ActivityToast.kt
package com.hjq.toast

import android.app.Activity
import android.view.View
import android.widget.TextView
import com.hjq.toast.config.IToast

class ActivityToast(activity: Activity) : IToast {
    private var mDuration: Int = 0
    private var mGravity: Int = 0
    private var mHorizontalMargin: Float = 0.0f
    private var mMessageView: TextView? = null
    private val mToastImpl: ToastImpl = ToastImpl(activity, this)
    private var mVerticalMargin: Float = 0.0f
    private var mView: View? = null
    private var mXOffset: Int = 0
    private var mYOffset: Int = 0

    override fun findMessageView(view: View): TextView {
        if (view is TextView) {
            val id = view.id
            if (id == View.NO_ID) {
                view.id = android.R.id.message
            } else if (id != android.R.id.message) {
                throw IllegalArgumentException("You must set the ID value of TextView to android.R.id.message")
            }
            return view
        }
        val messageView = view.findViewById<View>(android.R.id.message)
        if (messageView is TextView) {
            return messageView
        }
        throw IllegalArgumentException("You must include a TextView with an ID value of android.R.id.message")
    }

    override fun show() {
        mToastImpl.show()
    }

    override fun cancel() {
        mToastImpl.cancel()
    }

    override fun setText(textId: Int) {
        val view = mView
        if (view != null) {
            setText(view.resources.getString(textId))
        }
    }

    override fun setText(charSequence: CharSequence) {
        mMessageView?.text = charSequence
    }

    override fun setView(view: View?) {
        mView = view
        mMessageView = if (view == null) {
            null
        } else {
            findMessageView(view)
        }
    }

    override fun getView(): View? = mView

    override fun setDuration(duration: Int) {
        mDuration = duration
    }

    override fun getDuration(): Int = mDuration

    override fun setGravity(gravity: Int, xOffset: Int, yOffset: Int) {
        mGravity = gravity
        mXOffset = xOffset
        mYOffset = yOffset
    }

    override fun getGravity(): Int = mGravity

    override fun getXOffset(): Int = mXOffset

    override fun getYOffset(): Int = mYOffset

    override fun setMargin(horizontalMargin: Float, verticalMargin: Float) {
        mHorizontalMargin = horizontalMargin
        mVerticalMargin = verticalMargin
    }

    override fun getHorizontalMargin(): Float = mHorizontalMargin

    override fun getVerticalMargin(): Float = mVerticalMargin
}
